mage_integrations/mage_integrations/sources/sftp/tap_sftp/tap.py

```diff
@@ -0,0 +1,99 @@
+
+import singer
+from singer import metadata
+from tap_sftp import client
+from tap_sftp.discover import discover_streams
+from tap_sftp.stats import STATS
+from tap_sftp.sync import sync_stream
+from terminaltables import AsciiTable
+
+from mage_integrations.sources.base import write_schema, write_state
+
+REQUIRED_CONFIG_KEYS = ["username", "port", "host", "tables", "start_date"]
+REQUIRED_DECRYPT_CONFIG_KEYS = ['SSM_key_name', 'gnupghome', 'passphrase']
+REQUIRED_TABLE_SPEC_CONFIG_KEYS = ["delimiter", "table_name", "search_prefix",
+                                   "search_pattern"]
+LOGGER = singer.get_logger()
+
+
+def do_discover(config, logger=LOGGER):
+    logger.info("Starting discover")
+    streams = discover_streams(config)
+    if not streams:
+        raise Exception("No streams found")
+    catalog = {"streams": streams}
+    logger.info("Finished discover")
+    return catalog
+
+
+def stream_is_selected(mdata):
+    return mdata.get((), {}).get('selected', False)
+
+
+def do_sync(config, catalog, state, logger=LOGGER):
+    logger.info('Starting sync.')
+    sftp_client = client.connection(config)
+
+    for stream in catalog.streams:
+        stream_name = stream.tap_stream_id
+        mdata = metadata.to_map(stream.metadata)
+
+        if not stream_is_selected(mdata):
+            continue
+
+        write_state(state)
+        key_properties = metadata.get(metadata.to_map(stream.metadata),
+                                      (),
+                                      "table-key-properties")
+        replication_method = stream.replication_method
+        write_schema(stream_name,
+                     stream.schema.to_dict(),
+                     key_properties,
+                     replication_method=replication_method)
+
+        logger.info(f"{stream_name}: Starting sync")
+        counter_value = sync_stream(config, state, stream, sftp_client)
```

Comment:
Can we pass the logger to `sync_stream` and `sync_file` as well?

Reply:
Added on e690d19139e25188db4a840f2ba56ffc1575f7c0